example/dcgan/dcgan.go

```diff
@@ -8,6 +8,7 @@ import (
 	F "github.com/wangkuiyi/gotorch/nn/functional"
 	"github.com/wangkuiyi/gotorch/nn/initializer"
 	"github.com/wangkuiyi/gotorch/vision"
+	"github.com/wangkuiyi/gotorch/vision/mnist"
```

Comment:
I see the following PyTorch example on their API reference [page](https://pytorch.org/docs/stable/torchvision/datasets.html#mnist):

```python
imagenet_data = torchvision.datasets.ImageNet('path/to/imagenet_root/')
data_loader = torch.utils.data.DataLoader(imagenet_data,
                                          batch_size=4,
                                          shuffle=True,
                                          num_workers=args.nThreads)
```

Does it mean here we should use

```go
import "github.com/wangkuiyi/gotorch/vision/datasets"

imgnet := datasets.ImageNet()
```

And should we accept a little difference from PyTorch data loader creation API design -- to let the dataset object method create its loader?  I mean, in PyTorch, it is

```python
data_loader = torch.utils.data.DataLoader(imagenet_data, ...
```

In GoTorch, we could have

```go
dataloader = imgnet.NewLoader(...)
```

I propose this change because I noticed that the C++ loader is a template instantiation that is tightly bound to MNIST the dataset and is not independent of datasets as looks like the PyTorch API.

Reply:
I agree that we could have the  `Dataset` and `Dataloader` concept in GoTroch just like in PyTorch.
 
> dataloader = imgnet.NewLoader(...)

Maybe we can make  `Dataloader` as an independent implementation in GoTorch, that users only need to custom the new `Dataset` for a new dataset. In C++, users can also custom the dataset just like: https://github.com/pytorch/examples/blob/master/cpp/custom-dataset/custom-dataset.cpp


``` golang
imgnet := datasets.ImageNet("/yanxu/mnist", data.NoramlizerTransofrm(...))
dataloader := data.NewLoader(imanet, 32)
```